src/content-scripts/videoConsumption.content.js

```diff
@@ -0,0 +1,248 @@
+                    This happens when the video starts playing, and then the listeners try to attach to the in progress video */
+                if(vc.playbackStart === -1 && !video.paused) {
+                    console.log("Video (" + video.src + ") started (missed event)");
+                    vc.playbackStart = Date.now();
+                }
+            }             
+        }
+    }
+
+    /**
+     * Helper function that adds the amount of time that has elapsed since videoConsumption.playbackStart to the playbackDuration
+     */
+    function updatePlaybackDuration(videoConsumption) {
+        videoConsumption.playbackDuration += Date.now() - videoConsumption.playbackStart;
+    }
+
+    /**
+     * Helper function to update the playback duration for any videos that are still playing at the end of the page visit 
+     */
+    function cleanupPlayingVideos() {
+        for (let i = 0; i < videoConsumptions.length; i++) {
+            let vc = videoConsumptions[i];
+            if(!(vc.video.paused)) {
+                console.log("Stopping in progress video: " + vc.videoSrc);
+                updatePlaybackDuration(vc);
+            }
+        }
+    }
+
+    /**
+     * Helper function to aggregate the amount of time any video was playing on the page 
+     * @returns {number} the total number of milliseconds of video playback during a page visit from all videos on page
+     */
+    function aggregatePlaybackDuration() {
+        let playbackDuration = 0;
+        for(var i = 0; i < videoConsumptions.length; i++) {
+            playbackDuration += videoConsumptions[i].playbackDuration;
+        }
+        console.log("Total playback duration: " + playbackDuration);
+        return playbackDuration;
+    }
+
+    /**
+     * Helper function to parse the uploader channel's URL from the DOM of a YouTube page
+     * @returns {(string|boolean)} either a string of the URL or false if the DOM hasn't loaded that element yet
+     */
+    function getChannelURL() {
+        let currentHostname = (new URL(PageManager.url)).hostname;
+        if(currentHostname === "www.youtube.com") { // TODO: make this a youtube match pattern
+            /* Selector for the document element that provides a link to the uploader's YT channel */
+            let channelURLElement = document.querySelectorAll("#upload-info.ytd-video-owner-renderer .ytd-channel-name a");
+
+            /* If the element hasn't been populated yet, the query selector will return an empty array */
+            if(channelURLElement.length > 0) {
+                return channelURLElement[0].href;
+            }
+            else {
+                console.log("Video being played is on youtube, but the channel URL couldn't be found");
+                return "";
+            }
+        }
+    }
+
+    /**
+     * Helper function that transmits message with all information from page visit 
+     */
+    function sendVideoDataMessage(playbackDuration, timeStamp) {
```

Comment:
Could move this logic into the page visit stop handler, since that's the only place it's called.